chain/index/README.MD

```diff
@@ -0,0 +1,188 @@
+# ChainIndexer Documentation for RPC Providers
+
+## Introduction
+
+We're shipping a new Indexer implementation in Lotus (`ChainIndexer`) to index Filecoin chain state such as tipsets, messages, events and ETH transactions for reliable and faster RPC responses. The `ChainIndexer` replaces the existing `MsgIndex`, `EthTxIndex` and `EventIndex` implementations in Lotus which suffer from a multitude of known problems documented [here](https://github.com/filecoin-project/lotus/issues/12293). 
+
+**Note: If you are a Storage Provider or node operator who does not serve RPC requests, you can skip this document as the `ChainIndexer` is already disabled by default**. 
+
+This document is aimed at RPC providers and node operators who serve RPC requests and aims to walk through the configuration changes, migration flow and operations/maintenance work needed to enable, backfill and maintain the `ChainIndexer`. 
+
+## ChainIndexer Config
+
+The `ChainIndexer` must be enabled on an RPC node as it is disabled by default. Here is the mandatory config to use for all RPC providers to enable the `ChainIndexer` and for ensuring the `EthRPC` and `ActorEventsAPI` are enabled:
+
+```toml
+[ChainIndexer]
+# This is set to false by default which disables the ChainIndexer. 
+# Please ensure that you set it to true before starting your node.
+  EnableIndexer = true 
+
+[Fevm]
+# This is set to false by default which disables the ETH RPC API. 
+# Please ensure that you set it to true before starting your node.
+  EnableEthRPC = true
+
+[Events]
+# This is set to false by default which disables the Actor Events API. 
+# Please ensure that you set it to true before starting your node.
+  EnableActorEventsAPI = true
+```
+
+### Garbage Collection
+
+The `ChainIndexer` includes a garbage collection (GC) mechanism to manage the amount of historical data retained. By default, GC is disabled to preserve all indexed data.
+
+#### Configuration
+
+To configure GC, use the `GCRetentionEpochs` parameter in the `ChainIndexer` section of your config.
+
+The ChainIndexer periodically runs GC if `GCRetentionEpochs` is > 0 and removes indexed data for epochs older than `(current_head_height - GCRetentionEpochs)`.
```

Comment:
@BigLep No this isn't configurable right now. It's very easy to do so but I really want to ship with a minimal config here and then iterate based on user feedback rather than shipping a bloated config where most options are never used by users but do add to the complexity of the config.

It's a trivial change to make if we do get a strong ask for this.

Reply:
Understood that this isn't something we want to expose as configurable unless there is ask for it.  
And it sounds like we don't know if users care about how frequently it runs.
I think the path to take it to point to the source of truth.  In the updated revision I added this as a suggestion.  I'll resolve this conversation.